tests/all.rs

```diff
@@ -16,10 +16,11 @@ macro_rules! t {
 #[test]
 fn smoke_build_listener() {
     let b = t!(TcpBuilder::new_v4());
-    t!(b.bind("127.0.0.1:0"));
+    t!(b.bind("127.0.0.1:6543"));
```

Comment:
If we're going to hard-code the port, what about picking something a little higher (16384-32768) and not covered by an `/etc/services` listing?  It would give us a better chance of avoiding a clash with an existing thing, should the tests be run on a system with other running services.

Reply:
Failing on repeat runs (due to lingering sockets) seems less than ideal.  Skipping it probably makes sense for now.

Reply:
I think we're fine without it. Nothing tested this was correct before either. And I'm pretty confident it's correct if I can make it pass the `mio` and `tokio` test suites.